src/ukify/ukify.py

```diff
@@ -984,10 +930,47 @@ def make_uki(opts):
 """]
         uki.add_section(Section.create('.sbat', merge_sbat(input_pes, opts.sbat), measure=linux is not None))
 
+    # If we're building a UKI with additional profiles, the .profile section for the base profile has to be
+    # the last one so that everything before it is shared between profiles. The only thing we don't share
+    # between profiles is the .pcrsig section which is appended later and doesn't make sense to share.
+    if opts.profile and opts.join_profiles:
+        uki.add_section(Section.create(".profile", opts.profile, measure=True))
+
     # PCR measurement and signing
 
     call_systemd_measure(uki, opts=opts)
 
+    # UKI profiles
+
+    to_import = {'.linux', '.osrel', '.cmdline', '.initrd', '.ucode', '.splash', '.dtb', '.uname', '.sbat', '.profile'}
+
+    for profile in opts.join_profiles:
+        pe = pefile.PE(profile, fast_load=True)
+        prev_len = len(uki.sections)
+
+        names = [pe_strip_section_name(s.Name) for s in pe.sections]
+        names = [n for n in names if n in to_import]
+
+        if len(names) == 0:
```

Comment:
I think the explicit check is better and aligns more with systemd coding style

Reply:
Agreed that it does align more with the coding style, though I do find the shorter variant more readable, since it's closer to the human concept "if there are no names" than the C-style "if the length of the array containing names is zero".